qcodes/instrument_drivers/tektronix/AWG5014.py

```diff
@@ -1157,72 +1382,89 @@ def is_awg_ready(self):
             self.visa_handle.read()
         return True
 
-    def initialize_dc_waveforms(self):
-        self.set_runmode('CONT')
-        self.write('SOUR1:WAV "*DC"')
-        self.write('SOUR2:WAV "*DC"')
-        self.write('SOUR3:WAV "*DC"')
-        self.write('SOUR4:WAV "*DC"')
-        self.set_ch1_status('on')
-        self.set_ch2_status('on')
-        self.set_ch3_status('on')
-        self.set_ch4_status('on')
-
-    # QCodes specific parse_functions
-    def parse_int_pos_neg(self, val):
-        return ['POS', 'NEG'][val]
-
-    def parse_int_int_ext(self, val):
-        return ['INT', 'EXT'][val]
-
     def send_waveform_to_list(self, w, m1, m2, wfmname):
-        '''
-        Sends a complete waveform directly to the "User defined" waveform list. All parameters need to be specified.
-        See also: resend_waveform()
-
-        Input:
-            w (float[numpoints]) : waveform (must be a numpy array)
-            m1 (int[numpoints])  : marker1  (must be a numpy array)
-            m2 (int[numpoints])  : marker2  (must be a numpy array)
-            wfmname (string)    : waveform name
-            format (string):    'int' or 'real' (int has same awg output precision but much faster to transfer)
-        Output:
-            None
-        '''
+        """
+        Send a single complete waveform directly to the "User defined"
+        waveform list (prepend it). The data type of the input arrays
+        is unimportant, but the marker arrays must contain only 1's
+        and 0's.
+
+        Args:
+            w (numpy.ndarray): The waveform
+            m1 (numpy.ndarray): Marker1
+            m2 (numpy.ndarray): Marker2
+            wfmname (str): waveform name
+
+        Raises:
+            Exception: if the lengths of w, m1, and m2 don't match
+            TypeError: if the waveform contains values outside (-1, 1)
+            TypeError: if the markers contain values that are not 0 or 1
+        """
         logging.debug(
-            __name__ + ' : Sending waveform %s to instrument' % wfmname)
+            __name__ + ' : Sending waveform {} to instrument'.format(wfmname))
```

Comment:
Suggestion for logging:
use a module level logger as  in here:
https://github.com/QCoDeS/Qcodes/pull/416/files#diff-4c81a52a7bb6d513e5c5cd6c1b1f2c38R11

then all the calls to the logging system will include the module name which allows to tweak logging (say levels and handlers per module ) 👠